vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc

```diff
@@ -321,12 +326,15 @@ void UpholdWallet::OnTransferTokens(const type::Result result,
     return callback(type::Result::LEDGER_ERROR);
   }
 
-  if (uphold_wallet->status != type::WalletStatus::VERIFIED) {
+  if (uphold_wallet->status != type::WalletStatus::VERIFIED &&
+      uphold_wallet->status != type::WalletStatus::DISCONNECTED_VERIFIED) {
     return callback(type::Result::LEDGER_OK);
   }
 
-  DCHECK(!uphold_wallet->token.empty());
-  DCHECK(!uphold_wallet->address.empty());
+  if (uphold_wallet->status != type::WalletStatus::DISCONNECTED_VERIFIED) {
+    DCHECK(!uphold_wallet->token.empty());
+    DCHECK(!uphold_wallet->address.empty());
+  }
```

Comment:
Would you mind checking the invariant in the `type::WalletStatus::DISCONNECTED_VERIFIED` case, too?
Similarly to how we do it [here](https://github.com/brave/brave-core/blob/2b09a144e815d07ccc56241724f1deaacb7ad0ae/vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc#L92-L100).

Reply:
@szilardszaloki as we discussed on Slack, consolidated these checks into a separate function in [0e72501](https://github.com/brave/brave-core/pull/13037/commits/0e72501dcd4634aff39e6f46cfc9a859193a1ed7)